x-pack/plugins/security_solution/public/resolver/view/panels/use_formatted_date.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { i18n } from '@kbn/i18n';
+import moment from 'moment-timezone';
+import { useUiSetting } from '../../../../../../../src/plugins/kibana_react/public';
+
+const invalidDateText = i18n.translate(
+  'xpack.securitySolution.enpdoint.resolver.panelutils.invaliddate',
+  {
+    defaultMessage: 'Invalid Date',
+  }
+);
+
+/**
+ * Long formatter (to second) for DateTime
+ */
+export const formatter = new Intl.DateTimeFormat(i18n.getLocale(), {
+  year: 'numeric',
+  month: '2-digit',
+  day: '2-digit',
+  hour: '2-digit',
+  minute: '2-digit',
+  second: '2-digit',
+});
+
+/**
+ *
+ * @description formats a given time based on the user defined format in the advanced settings section of kibana under dateFormat
+ * @export
+ * @param {(ConstructorParameters<typeof Date>[0] | undefined)} timestamp
+ * @returns {(string | null)} - Either a formatted date or the text 'Invalid Date'
+ */
+export function useFormattedDate(
+  timestamp: ConstructorParameters<typeof Date>[0] | Date | undefined
+): string {
+  const dateFormatSetting: string = useUiSetting('dateFormat');
+  const timezoneSetting: string = useUiSetting('dateFormat:tz');
+
+  if (timestamp === undefined) return invalidDateText;
```

Comment:
Do we want to list the lack of a timestamp as an invalidDate or just leave it blank and return and empty string? There were some places where we did that, just wasn't sure if that would be confusing to the user. Maybe text here instead saying `Missing Time Details` or similar?

Reply:
ya, we could use the missing badge that we use somewhere else.

Reply:
Where can one find said badge?